Release 20211229.084554

jost-s opened the pull request.

steveej committed:
fix(scripts): proper architecture selection for nvfetcher

steveej committed:
fix(shell): don't pass 'inNixShell' on default.nix import

steveej commented:
my reproduction of the release steps in this repo produce a different result. here's the diff:

```diff
diff --git a/nix/nvfetcher/_sources/generated.nix b/nix/nvfetcher/_sources/generated.nix
index 6b52e18..90db2fa 100644
--- a/nix/nvfetcher/_sources/generated.nix
+++ b/nix/nvfetcher/_sources/generated.nix
@@ -201,12 +201,12 @@
       fetchSubmodules = false;
       deepClone = false;
       leaveDotGit = false;
-      sha256 = "sha256-3udcqyCkpGbJYDc0rxBND7gpV/XAKR8e7WyhEThVaUE=";
+      sha256 = "0hb9alw138bcxlg1yaf0ymbjkf0g9l8ayd1pc34nd95442mmrryy";
     };
     cargoLock = {
       lockFile = ./holochain_revision_holochain-0_0_122-holochain-0.0.122/Cargo.lock;
       outputHashes = {
-        "cargo-test-macro-0.1.0" = "sha256-hIGpT0n41CA24vss4itXS3O2XrznsBce/60PUVrwwfs=";
+        "cargo-test-macro-0.1.0" = "1yy1y1d523xdzwg1gc77pigbcwsbawmy4b7vw8v21m7q957sk0c4";
       };
     };
   };
diff --git a/packages/holochain/versions/v0_0_122.nix b/packages/holochain/versions/v0_0_122.nix
index c92c481..f8b699a 100644
--- a/packages/holochain/versions/v0_0_122.nix
+++ b/packages/holochain/versions/v0_0_122.nix
@@ -1,14 +1,14 @@
 # This file was generated with the following command:
-# /nix/store/4hg1qhfyykzygzvxm2y3gq60331slwjj-update-holochain-versions/bin/update-holochain-versions --nvfetcher-dir=nix/nvfetcher --output-file=packages/holochain/versions/v0_0_122.nix --git-src=revision:holochain-0.0.122 --lair-version-req=~0.0
+# /nix/store/dvn8q4vc9vd0gzzdbhsipfp2iamd9vwj-update-holochain-versions/bin/update-holochain-versions --nvfetcher-dir=nix/nvfetcher --output-file=packages/holochain/versions/v0_0_122.nix --git-src=revision:holochain-0.0.122 --lair-version-req=~0.0
 # For usage instructions please visit https://github.com/holochain/holochain-nixpkgs/#readme

 {
     url = "https://github.com/holochain/holochain";
     rev = "holochain-0.0.122";
-    sha256 = "sha256-3udcqyCkpGbJYDc0rxBND7gpV/XAKR8e7WyhEThVaUE&#x3D;";
+    sha256 = "0hb9alw138bcxlg1yaf0ymbjkf0g9l8ayd1pc34nd95442mmrryy";
     cargoLock = {
         outputHashes = {
-            "cargo-test-macro-0.1.0" = "sha256-hIGpT0n41CA24vss4itXS3O2XrznsBce/60PUVrwwfs&#x3D;";
+            "cargo-test-macro-0.1.0" = "1yy1y1d523xdzwg1gc77pigbcwsbawmy4b7vw8v21m7q957sk0c4";
         };
     };

@@ -21,7 +21,7 @@
     lair = {
         url = "https://github.com/holochain/lair";
         rev = "v0.0.9";
-        sha256 = "sha256-glSixh2GtWtJ1wswAA0Q2hnLIFPQY+Tsh36IcUgIbRs&#x3D;";
+        sha256 = "06vd1147323yhznf8qyhachcn6fs206h0c0bsx4npdc63p3a4m42";

         binsFilter = [
             "lair-keystore"
```

jost-s commented:
Mmh, hashes are different and on your machine there weren't any `sha256-` prefixes. Okay, let's see what that's about.

steveej committed:
nix/shell: add shell flavors and use nixpkgs-fmt

steveej committed:
overlays/rust: add darwin target

steveej committed:
update-holochain-versions: refactor, test and bugfix version rendering

steveej committed:
github: run `cargo test` on CI

steveej committed:
update generated crate expressions

steveej committed:
update-holochain-versions: don't html-escape any hash value

steveej committed:
nix-shell: use nixUnstable

steveej committed:
incomplete: stub out running the crate tests in nix

steveej committed:
CI: stop building holochain 0.0.{110,115,116}

steveej committed:
scripts: improve hnixkpgs-update-all and add single version

steveej committed:
update-holochain-versions: serialize args if it exists, gracefully handle otherwise

steveej committed:
update-holochain-versions: pass `--filter` to nvfetcher on update for minimal changes

steveej committed:
unignore nvfetcher state

steveej committed:
update script: improve git commit message

steveej committed:
update-holochain-versions: only encode filename of binary in the version file comments

steveej committed:
track shake.database with git lfs

steveej committed:
github: disable LFS fetching on CI as we don't need it there yet

steveej committed:
scripts: include shake files in diff targets

steveej committed:
improve shell structure and configure nixUnstable

steveej committed:
script: seperate commit for nvfetcher state file

steveej committed:
update nvfetcher state

steveej committed:
update nvfetcher sources

the following keys were updated

develop develop_lair_0_1 main v0_0_110 v0_0_115 v0_0_116 v0_0_117 v0_0_118 v0_0_119 v0_0_120 v0_0_121

jost-s committed:
add holochain-0.0.122

jost-s committed:
fix experimental feature flag in scrips.nix

jost-s committed:
update nvfetcher sources

the following keys were updated

v0_0_122

jost-s committed:
update nvfetcher state

jost-s committed:
fix pkgs variable in scripts.nix

jost-s commented:
Redoing the whole release after fixing issues.
